samples/Samples.TracingWithoutLimits/Program.cs

```diff
@@ -0,0 +1,200 @@
+
+        private static void Main()
+        {
+            var total = 500;
+
+            PrepKeys(ServiceDogWalker, RootWalkOperation, total * 1m);
+            PrepKeys(ServiceCatWalker, RootWalkOperation, total * 0.8m);
+            PrepKeys(ServiceRatWalker, RootWalkOperation, total * 0.5m);
+            PrepKeys(ServiceDogRunner, RootRunOperation, total * 0.2m);
+            PrepKeys(ServiceCatRunner, RootRunOperation, total * 0.1m);
+            PrepKeys(ServiceRatRunner, RootRunOperation, total * 0.0m);
+
+            PrepKeys(UnknownService, UnknownOperation, null);
+
+            // Configured rules:
+            // [{"service":"rat.*","name":".*run.*","sample_rate":0},{"service":"dog.*","name":".+walk","sample_rate":1.0},{"service":"cat.*","name":".+walk","sample_rate":0.8},{"name":".+walk","sample_rate":0.5},{"service":"dog.*","sample_rate":0.2},{"service":"cat.*","sample_rate":0.1}]
+            for (var i = 1; i <= total; i++)
+            {
+                Console.WriteLine($"{ConsoleTime()} Iteration {i} beginning.");
+
+                // We should keep 100 %
+                RunStuff(ServiceDogWalker, RootWalkOperation);
+
+                // We should keep 80%
+                RunStuff(ServiceCatWalker, RootWalkOperation);
+
+                // We should keep 50%, because it matches the fallback ".+walk" rule
+                RunStuff(ServiceRatWalker, RootWalkOperation);
+
+                // We should keep 20%, as it matches the fallback dog rule
+                RunStuff(ServiceDogRunner, RootRunOperation);
+
+                // We should keep 10%, as it matches the fallback cat rule
+                RunStuff(ServiceCatRunner, RootRunOperation);
+
+                // We should keep 0%, we don't want running rats
+                RunStuff(ServiceRatRunner, RootRunOperation);
+
+                // This should fallback to the default behavior, determined by the agent
+                RunStuff(UnknownService, UnknownOperation);
+
+                Console.WriteLine($"{ConsoleTime()} Iteration {i} ending.");
+            }
+
+            Console.WriteLine();
+
+            foreach (var key in Counts.Keys)
+            {
+                var isExpect = key.Contains("Expecting");
+
+                if (isExpect)
+                {
+                    Console.WriteLine();
+                    Console.WriteLine($"{key}");
+                }
+                else
+                {
+                    Console.WriteLine($"{key}: {Counts[key]}");
+                }
+            }
+        }
+
+        private static string ConsoleTime()
+        {
+            return $"[{DateTime.Now:hh:mm:ss:fff}]";
+        }
+
+        private static void RunStuff(string serviceName, string operationName)
+        {
+            Counts[Key(serviceName, operationName)]++;
+
+            Scope root;
+
+            using (root = Tracer.Instance.StartActive(operationName: operationName, serviceName: serviceName))
```

Comment:
Tests that only use manual instrumentation don't need an additional project, since we don't need to attach the profiler.